test/eth/TransactionManager.spec.js

```diff
@@ -151,8 +151,8 @@ test('lifecycle hooks', async () => {
   await Promise.all([lock, mineBlocks(service)]);
 
   // deposit, approve WETH, join, approve PETH, lock
```

Comment:
I had to change the expects here for it to pass, I thought it was because I was calling `lockEth` in my test below, but then wouldn't that increase the number of times the handlers were called?

Reply:
AllowanceService.requireAllowance is called twice, once for WETH and once for PETH. If it determines that the allowance is already high enough, it won't create a new approvals transaction. So this was failing for you because at some point the test chain didn't get reverted, so the approvals stuck around and affected later test runs. So restarting the testchain with `--snapshot default` should fix it. And generally to prevent it from happening, make sure Jest exits cleanly (i.e. with the "q" command during watch mode, or by finishing completely in non-watch mode).